### ISSUE TITLE
Should the parser calculate the data type for expressions in advance as part of parsing validation? Can it?

### ISSUE THREAD

**Issue by aszs** [oasis-open/tosca-community-contributions#70]

**Comment by philippemerle** [comment]
> Can it?

My TOSCA parser is able to compute the type of expressions such as get_property, get_attribute, etc. Then typing errors can be detected.

**Comment by lauwers** [comment]
Yes, my parser implements the same behavior: it checks if the type of the value referenced by 'get_property' or 'get_attribute' calls matches the type of the parameter definition that uses the 'get_property' or 'get_attribute' call.
